tasks/pkg-suse.yml

```diff
@@ -0,0 +1,59 @@
+---
+- block: # Work around due to SNI check for SLES11
+  - name: Stat if RPM key already exist
+    stat:
+      path: /tmp/DATADOG_RPM_KEY.public
+    register: ddkey
+  - name: Download new RPM key
+    shell: "curl {{ datadog_zypper_gpgkey }} -o /tmp/DATADOG_RPM_KEY.public"
+    args:
+      warn: no
+    when: not ddkey.stat.exists
+  when: ansible_distribution_version|int == 11
+
+- name: Download new RPM key
+  get_url:
+    url: "{{ datadog_zypper_gpgkey }}"
+    dest: /tmp/DATADOG_RPM_KEY.public
+    sha256sum: "{{ datadog_zypper_gpgkey_sha256sum }}"
+  when: ansible_distribution_version|int >= 12
+
+- name: Import new RPM key
+  rpm_key:
+    key: /tmp/DATADOG_RPM_KEY.public
+    state: present
+  when: not ansible_check_mode
+
+- name: Install DataDog zypper repo
+  zypper_repository:
+    name: datadog
+    description: Datadog, Inc.
+    repo: "{{ datadog_zypper_repo }}"
+    enabled: yes
+    auto_import_keys: yes
+    disable_gpg_check: "{% if ansible_distribution_version|int == 11 %}yes{% else %}no{% endif %}" # SLES11 considers datadog's repo as untrusted
```

Comment:
I tried the role on suse 12, and I get an error from this. Sadly ansible is too limited to allow us to set the `repo_gpgcheck=0` option, we only have acces to `gpgcheck`. I don't know how I feel about disabling gpg check for every version of SUSE.
Did you achieve to make it work on SUSE >= 12 with the GPG check activated ?

Worst case could be to create the repository file manually with a template file.
The version on the official install page on datadog.com is:
```
[datadog]
name=Datadog, Inc.
enabled=1
baseurl=https://yum.datadoghq.com/suse/stable/6/x86_64
type=rpm-md
gpgcheck=1
repo_gpgcheck=0
gpgkey=https://yum.datadoghq.com/DATADOG_RPM_KEY.public
```

Reply:
Zypper has two different types of gpg checks, the repo one and the other one. If you're importing the gpg key properly, it should work for packages. (I don't know ansible well enough to know if t's being imported properly.

Reply:
@hush-hush and @gmmeyer, Zypper not importing GPG keys properly may be the same issue filed with OpenSuse, see https://github.com/openSUSE/zypper/issues/144

Let me know if you are okay with gpgcheck disabled with SLES11 OS.

Thanks